### DISCUSSION TITLE
Getting 422 from axios post request when using Pydantic models

### DISCUSSION THREAD

**Discussion by PauloFavero** [fastapi/fastapi#6336] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Request
from fastapi.middleware.cors import CORSMiddleware
from typing import Dict, Optional
from pydantic import BaseModel

allowed_origins = ["http://localhost:3344", "http://localhost:3003"]
allowed_methods = ["POST, GET"]

class Getters(BaseModel):
    acqCtrl: Optional[str]
    imgCtrl: Optional[str]
    algCtrl: Optional[str]

class Config(BaseModel):
    serial: str
    get: Getters


server = FastAPI()

server.add_middleware(
    CORSMiddleware,
    allow_origins=allowed_origins,
    allow_credentials=False,
    allow_methods=allowed_methods,
    allow_headers=["*"],
)

@server.post("/config")
def read_item(data: Config):
    print(data.dict())
    return {"ok": "ok"}
```

### Description

I launch the server and then I make requests using swagger. Everything works fine but when I use my client that is making request with axios I always get 422 Unprocessable entity. 

The object that I am sending is the following:

```typescript
const config = {
  "serial": "string",
  "get": {
    "acqCtrl": "test",
    "imgCtrl": "test2",
    "algCtrl": "test3",
   }
}
```

On the client side, axios is making the post request as follows:

```typescript
export class AxiosHttpClient implements HttpClient {
  async request (data: HttpRequest): Promise<HttpResponse> {
    let axiosResponse: AxiosResponse
    try {
      axiosResponse = await axios.request({
        url: data.url,
        method: 'post',
        data: data.body // config is in data.body
        headers: { 'Content-Type': 'application/json' }
      })
    } catch (error) {
      console.log('Axios response', axiosResponse)
      console.log('Axios error', error)
      axiosResponse = error.response
    }
    return {
      statusCode: axiosResponse.status,
      body: axiosResponse.data
    }
  }
}
```
The error is as follows:

```
axios error [
{
 loc: [ 'body', 'get', 'acqCtrl', 'imgCtrl', 'algCtrl' ],
  msg: 'value is not a valid dict',
type: 'type_error.dict'
}
 ]
```

The versions being used in the project are: 

fastapi==0.74.1
uvicorn==0.17.5
pydantic==1.9.0
"axios": "^0.24.0"

It's important to note that if, I use Request instead of Pydantic models everything works fine.

### Operating System

Linux, Other

### Operating System Details

I am using docker containers

### FastAPI Version

0.74.1

### Python Version

3.8.7

**Comment by PauloFavero** [comment, marked as answer]
Sorry about that! The issue was related to the client. Everything is working great!

**Reply by felexkemboi** [reply]
Could you please explain how what was the issue with your Client

**Reply by alejsdev** [reply]
Thanks for reporting back! @PauloFavero

**Comment by ncolterx** [comment]
> Sorry about that! The issue was related to the client. Everything is working great!

Do you have any details on what the issue actually was?

**Reply by yashkashustrij** [reply]
Yeah, you may try this:

https://axios-http.com/docs/urlencoded
